lib/sinatra/base.rb

```diff
@@ -1029,7 +1029,7 @@ def process_route(pattern, conditions, block = nil, values = [])
       if regexp_exists
         captures           = pattern.match(route).captures
         values            += captures
-        @params[:captures] = captures
+        @params[:captures] = captures if !captures.nil? && !captures.empty?
```

Comment:
If you hope that the captures checking is using `Object#nil?`, you could simplify the condition by using De Morgan's laws.
`unless captures.nil? || captures.empty?`

Reply:
refactored and fixup in this commit https://github.com/sinatra/sinatra/pull/1390/commits/51fe0fcb00cf8f21c524370a3b7c7a4a4c7d067a